ethcore/engine/src/engine.rs

```diff
@@ -295,6 +295,9 @@ pub trait Engine: Sync + Send {
 	/// Register a component which signs consensus messages.
 	fn set_signer(&self, _signer: Box<dyn EngineSigner>) {}
 
+	/// Unregisters the engine signer address to stop signing consensus messages.
+	fn clear_signer(&self) {}
```

Comment:
I wish there was a way to achieve this without burdening the `Engine` trait further. I fear it would require quite a few changes though. :/

Reply:
What if we could use the `set_signer()` method for this instead? If the `signer` argument were an `Option<Box<dyn EngineSigner>>` couldn't we move the logic for clearing the signer to the engine impl?